scripts/pyenv_setup.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# This script correctly sets up pyenv
+#
+# Assumptions:
+# - This script assumes you're calling from the top directory of the repository
+set -eu
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+query_big_sur() {
+    if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+get_shell_startup_script() {
+  if [ -n "$SHELL" ]; then
+    case "$SHELL" in
+      /bin/bash)
```

Comment:
This isn't good, because there aren't any canonical paths a shell could be in. It could be in `/bin`, `/sbin`, `/usr/local/bin`, etc. You should strip away all path components but leave the last / in case there's some weird shell like yzsh, lol. So like:

```
case "$SHELL" in
  */bash)
    echo bash;;
...
```

Reply:
Thank you for catching this!